src/kw_env.sh

```diff
@@ -303,30 +314,37 @@ function list_env_available_envs()
   local local_kw_configs="${PWD}/.kw"
   local current_env
   local output
-  local ret
+  declare -a all_envs
 
   if [[ ! -d "$local_kw_configs" ]]; then
     complain 'It looks like that you did not setup kw in this repository.'
     complain 'For the first setup, take a look at: kw init --help'
     exit 22 # EINVAL
   fi
 
-  output=$(ls "${local_kw_configs}/${ENV_DIR}" 2> /dev/null)
-  ret=$?
-  if [[ ret -ne 0 || -z $output ]]; then
-    say 'Kw did not find any environment. You can create a new one with the --create option.'
-    say 'See kw env --help'
+  if [[ ! -d "${local_kw_configs}/${ENV_DIR}" ]]; then
+    no_env_message
+    return 0
+  fi
+
+  readarray -t all_envs < <(find "${local_kw_configs}/${ENV_DIR}" -maxdepth 1 -type d -printf '%P\n' | sort --dictionary-order)
+  if [[ "${#all_envs[@]}" -eq 0 ]]; then
+    no_env_message
     return 0
   fi
 
   if [[ -f "${local_kw_configs}/${ENV_CURRENT_FILE}" ]]; then
     current_env=$(< "${local_kw_configs}/${ENV_CURRENT_FILE}")
-    say "Current env:"
-    printf ' -> %s\n' "$current_env"
+    say 'Current env:'
+    printf ' -> %s: %s\n\n' "$current_env" "${KW_CACHE_DIR}/${ENV_DIR}/${current_env}"
```

Comment:
When testing, the path `"${KW_CACHE_DIR}/${ENV_DIR}/${current_env}"`, points only to the directory where the environment `.config` is stored. FWIU, the env comprises the kernel `.config` plus the kw configs (stored at `.kw/envs`).

I get that the kw configs location may be trivial, as `kw env` doesn't have local scopes (i.e., the kw configs listed will always be inside `${PWD}/.kw/envs`), but is the kernel `.config` file sufficient? If so, ignore this comment.

![Screenshot from 2024-02-12 20-24-09](https://github.com/kworkflow/kworkflow/assets/95055784/ffb35565-b8dd-44cf-b483-c9cfc1552b45)

Reply:
From your update, you got the idea, but just to give you one context, I usually want to run the below command for running the unit test:

```
./tools/testing/kunit/kunit.py run --arch=x86_64 --build_dir=/home/siqueira/.cache/kw/envs/X86_64_KUNIT  --kunitconfig=drivers/gpu/drm/amd/display/test/kunit
```

However, I constantly forget the cache path. This improvement in the list feature will help me and others with that.

Reply:
Thanks for the context. It makes even clearer now :+1: